### PR TITLE
Add preliminary torch implementation of LIF neuron

### PR DESCRIPTION
**Pull request by cpehle**
This implements a (simple) LIF neuron model integrated into the machinery of pytorch, the backward 
static method most likely does not work in the current form, instead it is here where one could experiment with different models for the gradient.

### CODE REVIEW

File: volrpynn/snn_test.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    m = snn.LIF(1,1)
+    m.weight.data = torch.tensor([[3.0]])
+
+    s = m(x)
+
+    assert(torch.sum(s).item() == 7)
+
+
+if __name__ == "__main__":
+    test_default_parameters()
```
Comment (Jegp) [inline volrpynn/snn_test.py:167]: This can be autodiscovered with pytest

File: volrpynn/snn.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+class LIFTransferFunction(torch.autograd.function.Function):
+    """Implements a lif transfer function.
+
+    Note: Uses hand rolled euler integration, might be a better idea to use an existing
+    nest/genn implementation.
+    """
+
+    @staticmethod
+    def forward(ctx, input, weight, lif_parameters, num_timesteps, dt):
```
Comment (Jegp) [inline volrpynn/snn.py:143]: It's possible to use `input` as a name, but it's also a reserved keyword...

File: volrpynn/snn.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        if ctx.needs_input_grad[0]:
+            grad_input = grad_output.mm(weight)
+        if ctx.needs_input_grad[1]:
+            grad_weight = grad_output.t().mm(input)
+
+        return grad_input, grad_weight, grad_voltages, grad_spikes, grad_lif_parameters, grad_num_timesteps, grad_dt
+
+lif_transfer_function = LIFTransferFunction.apply
+
+class LIF(torch.nn.Module):
```
Comment (Jegp) [inline volrpynn/snn.py:167]: Cool concept. This could be the more general 'wrapper' class you talked about. And then we could abstract the implementation away.

**Review by Jegp — APPROVED** [review]
